rustworkx-core/src/generators/simple_shortest_paths.rs

```diff
@@ -0,0 +1,368 @@
+        }
+    }
+}
+
+// This is mutation of petgraph dijkastra to get full path between source to target
+fn dijkstra<G, F, K>(
+    graph: G,
+    start: G::NodeId,
+    goal: Option<G::NodeId>,
+    mut edge_cost: F,
+) -> (HashMap<G::NodeId, K>, HashMap<G::NodeId,G::NodeId>)
+where
+    G: IntoEdges + Visitable,
+    G::NodeId: Eq + Hash,
+    F: FnMut(G::EdgeRef) -> K,
+    K: Measure + Copy,
+    <G>::NodeId: Debug,
+{
+    let mut visited = graph.visit_map();
+    let mut scores = HashMap::new();
+    let mut visit_next = BinaryHeap::new();
+    let zero_score = K::default();
+    scores.insert(start, zero_score);
+    let mut tracing: HashMap<G::NodeId, G::NodeId> = HashMap::new();
+    visit_next.push(MinScored(zero_score, start));
+    while let Some(MinScored(node_score, node)) = visit_next.pop() {
+        if visited.is_visited(&node) {
+            continue;
+        }
+        if goal.as_ref() == Some(&node) {
+            break;
+        }
+        for edge in graph.edges(node) {
+            let next = edge.target();
+            if visited.is_visited(&next) {
+                continue;
+            }
+            let next_score = node_score + edge_cost(edge);
+            match scores.entry(next) {
+                Occupied(ent) => {
+                    if next_score < *ent.get() {
+                        *ent.into_mut() = next_score;
+                        visit_next.push(MinScored(next_score, next));
+                        if let Some(v) = tracing.get_mut(&next) {
+                            *v = node;
+                        } else {
+                            tracing.insert(next, node);
+                            todo!()
+                        };
+                        
+                    }
+                }
+                Vacant(ent) => {
+                    ent.insert(next_score);
+                    visit_next.push(MinScored(next_score, next));
+                    tracing.insert(next, node);
+                }
+            }
+        }
+        visited.visit(node);
+    }
+    (scores, tracing)
+}
+
+// This function is private to this module, will call Dijkstra algo to get the possible path & Scores & returns a SimplePath as return value
+
+fn get_simple_path(s: &mut SimplePath) -> Option<SimplePath> {
+    let (score, path) = dijkstra(&s.graph, s.source, Some(s.target), |e| *e.weight());
+    let mut score_target: f32 = 0.0;
+    let mut unique_paths = s.unique_paths.clone();
+    let mut paths :Vec<NodeIndex> = vec!();
+
+    if score.contains_key(&s.target) {
+        score_target = *score.get(&s.target).expect("Error");
+    }
+    
+    if path.contains_key(&s.target)  {
+        paths.push(s.target);
+        let mut node = &s.target;
+        loop { 
+            let pre_node = path.get(node).expect("Error");
+            paths.push(*pre_node);
+            if *pre_node == s.source { 
+                break;
+            }
+            node = pre_node;
+            }
```

Comment:
In general it'd be better to rework this as a while loop so the exit conditions are a bit more clear.

Reply:
it actually back tracks the path till source, if the next node is source, then it exits.

> In general it'd be better to rework this as a while loop so the exit conditions are a bit more clear.

it actually back tracks the path till source, if the next node is source, then it exits.

Reply:
see the current status, even though its same with comment one will get to know the exit state.